css/settings/_variables.scss

```diff
@@ -6,7 +6,10 @@ $asset-path: "../";
 $font-path: "#{$asset-path}fonts";
 $image-path: "#{$asset-path}img";
 
-$external-href: '[href^="http"]:not([href*="designsystem.digital.gov"])';
+$touchpoints-href: '[href^="https://touchpoints.app.cloud.gov/touchpoints"]';
+$federalist-href: '[href^="https://federalist"]';
+$site-href: '[href*="designsystem.digital.gov"]';
+$external-href: '[href^="http"]:not(#{$site-href}, #{$touchpoints-href}, #{$federalist-href})';
```

Comment:
These seem to work as expected, but I am curious about using both `href*=` and `href^=` here.

Reply:
It would be more manageable long term to check for all `https` that have `.gov`.


```suggestion
$gov-href: '[href*=".gov"]';
$external-href: '[href^="https://"]:not(#{$gov-href})';
```

Reply:
## Using both `href*=` and `href^=`

I decided to use each of these selectors to be more definitive of what I'm targeting. I believe we could potentially use `*=` for all of them, but the prefix selector seemed to be appropriate in my opinion.

`href*=` - Checks for value anywhere within the `href` value
> Represents elements with an attribute name of attr whose value contains at least one occurrence of value within the string. [^1]

`href^=` - This specifically checks the beginning of the `href` value
> Represents elements with an attribute name of attr whose value is prefixed (preceded) by value. [^2]

[^1]: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_6

[^2]: https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#attrvalue_4

## Targetting non `.gov` links

Implementing @mejiaj's suggestion causes a visual regression when linking to `.gov` sites that aren't the design system page

![image](https://github.com/uswds/uswds-site/assets/25211650/b2e7b334-0315-432e-a6b0-5250200e4d69)

The goal with my approach is to add the external link styles to all links that are not:

- The USWDS Site
- The touchpoints survey (visual regression)
- A link to a component/template demo page (visual regression)

All other external links should maintain the external link styles.

We _should_ update the Federalist link to `pages.cloud.gov` so that other links to `federalist` sites still maintain the external link styles. The Federalist link on the implementations page had unintentionally lost its icon.

Updated in [e0ffd5c](https://github.com/uswds/uswds-site/pull/2318/commits/e0ffd5c6a474212fc061c220a1e04fd23e70e77b)